src/main/java/uk/gov/pay/ledger/transaction/service/TransactionService.java

```diff
@@ -88,9 +89,10 @@ public TransactionSearchResponse searchTransactions(TransactionSearchParams sear
         return searchTransactions(null, searchParams, uriInfo);
     }
 
-    public TransactionSearchResponse searchTransactions(String gatewayAccountId, TransactionSearchParams searchParams, UriInfo uriInfo) {
-        if (StringUtils.isNotBlank(gatewayAccountId)) {
-            searchParams.setAccountId(gatewayAccountId);
+    public TransactionSearchResponse searchTransactions(List<String> gatewayAccountIds,
+                                                        TransactionSearchParams searchParams, UriInfo uriInfo) {
+        if (!isEmpty(gatewayAccountIds)) {
```

Comment:
If the callers of this were just changed so that they pass an empty list rather than `null`, this could just be `gatewayAccountIds.isEmpty()`.